internal/tsetse/rules/ban_nested_it.ts

```diff
@@ -0,0 +1,57 @@
+/**
+ * @fileoverview Bans nested 'it' methods.
+ * As these cause falsy positive in unit tests.
+ */
+
+import * as ts from 'typescript';
+
+import { Checker } from '../checker';
+import { ErrorCode } from '../error_code';
+import { AbstractRule } from '../rule';
+
+const FAILURE_STRING = `Nested 'it' are not allowed. Did you mean to use 'describe'?`;
+
+export class Rule extends AbstractRule {
+	readonly ruleName = 'ban_nested_it';
+	readonly code = ErrorCode.BAN_NESTED_IT;
+
+	register(checker: Checker) {
+		checker.on(ts.SyntaxKind.CallExpression, checkCallExpression, this.code);
+	}
+}
+
+function checkCallExpression(checker: Checker, node: ts.CallExpression) {
+	const identifier = (node.expression as ts.Identifier);
+	if (identifier.getText() === 'it' && hasNestedItExpression(node)) {
```

Comment:
I think we should resolve the symbol to which module defines it - otherwise someone writing their own method called `it` would have this check trigger

Reply:
Yeah good point. Give me some time and I'll do it.

Reply:
Do you know any other way to get the `moduleName` other than

```ts
const signature = checker.typeChecker.getResolvedSignature(node);
const file = signature.getDeclaration().getSourceFile().moduleName;
```

As the moduleName here it always being returned `undefined`, but If I try to use `.fileName` instead I get the correct file path. 

Cause I was not going to do something like;

```ts
const signature = checker.typeChecker.getResolvedSignature(node);
const file = signature.getDeclaration().getSourceFile().fileName;

if (file.indexOf('jasmine') >= 0  && node.getText() === 'it') 
```

I don't like the indexOf implementation that much, but cannot find a way to get the moduleName since it's being returned undefined.

Reply:
This has been addressed as well